azurerm/internal/services/postgres/postgresql_flexible_server_database_resource_test.go

```diff
@@ -0,0 +1,116 @@
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("charset").HasValue("UTF8"),
+				check.That(data.ResourceName).Key("collation").HasValue("en_US.UTF8"),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccPostgresqlFlexibleServerDatabase_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_postgresql_flexible_server_database", "test")
+	r := PostgresqlFlexibleServerDatabaseResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func TestAccPostgresqlFlexibleServerDatabase_charsetLowercase(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_postgresql_flexible_server_database", "test")
+	r := PostgresqlFlexibleServerDatabaseResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.charsetLowercase(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("charset").HasValue("LATIN1"),
+				check.That(data.ResourceName).Key("collation").HasValue("en_US.latin1"),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func (PostgresqlFlexibleServerDatabaseResource) Exists(ctx context.Context, clients *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := parse.FlexibleServerDatabaseID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := clients.Postgres.FlexibleServerDatabaseClient.Get(ctx, id.ResourceGroup, id.FlexibleServerName, id.DatabaseName)
+	if err != nil {
+		return nil, fmt.Errorf("retrieving Azure Postgresql Flexible Server database %q on server %q (Resource Group: %q): %+v", id.DatabaseName, id.FlexibleServerName, id.ResourceGroup, err)
+	}
+
+	return utils.Bool(resp.DatabaseProperties != nil), nil
+}
+
+func (r PostgresqlFlexibleServerDatabaseResource) requiresImport(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_postgresql_flexible_server_database" "import" {
+  name      = azurerm_postgresql_flexible_server_database.test.name
+  server_id = azurerm_postgresql_flexible_server_database.test.server_id
+  collation = azurerm_postgresql_flexible_server_database.test.collation
+  charset   = azurerm_postgresql_flexible_server_database.test.charset
+}
+`, r.basic(data))
+}
+
+func (PostgresqlFlexibleServerDatabaseResource) charsetLowercase(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_postgresql_flexible_server_database" "test" {
+  name      = "acctest-fsd-%d"
+  server_id = azurerm_postgresql_flexible_server.test.id
+  collation = "en_US.latin1"
+  charset   = "latin1"
+}
+`, PostgresqlFlexibleServerResource{}.basic(data), data.RandomInteger)
+}
```

Comment:
fwiw this can become:

```suggestion
func (r PostgresqlFlexibleServerDatabaseResource) charsetLowercase(data acceptance.TestData) string {
	return fmt.Sprintf(`
%s

resource "azurerm_postgresql_flexible_server_database" "test" {
  name      = azurerm_postgresql_flexible_server.test.name
  server_id = azurerm_postgresql_flexible_server.test.id
  collation = azurerm_postgresql_flexible_server.test.collation
  charset   = azurerm_postgresql_flexible_server.test.charset
}
`, r.basic(data))
}
```

Reply:
This won't work we need PostgresqlFlexibleServerResource for base resources.